src/geometry.ts

```diff
@@ -0,0 +1,60 @@
+export type Rect = {
+  left: number;
+  top: number;
+  width: number;
+  height: number;
+};
+
+export type Point = {
+  x: number;
+  y: number;
+};
+
+/// Measures from some box (or point) outwards
+export type MarginBox = {
+  top: number;
+  right: number;
+  bottom: number;
+  left: number;
+};
+
+/// Measures from some box inwards
+export type PaddingBox = {
+  top: number;
+  right: number;
+  bottom: number;
+  left: number;
+};
```

Comment:
@birtles Is there a reason for the redundancy with Rect here? I'm guessing for potentially better Intellisense, but if not, we might as well alias these.

```suggestion
/// Measures from some box (or point) outwards
export type MarginBox = Rect;

/// Measures from some box inwards
export type PaddingBox = Rect;
```

Reply:
I think the only difference is that `Rect` uses `width`/`height` whereas the two boxes use `bottom`/`right`. We could definitely alias, e.g. `PaddingBox` to `MarginBox` though or just conflate them into one.

(`DOMRect` actually has both `width`/`height` and `bottom`/`right` from memory but it uses `bottom`/`right` in a different way to us.)

Reply:
Oh whoops, I misread `Rect`. How about this, then:

```suggestion
export type Box = {
  top: number;
  right: number;
  bottom: number;
  left: number;
};

/// Measures from some box (or point) outwards
export type MarginBox = Box;

/// Measures from some box inwards
export type PaddingBox = Box;
```